Add a setup script to pull all available API data for development

craig-cogdill opened the pull request:
A new script, `scripts/setupDev.js` will pull the `states.json` and all available `states/<STATE>/<STATE>.json` data for every state.

I've also added new yarn commands to the `package.json` file and updated the README with the new instructions for getting a basic development environment started.

AnEmortalKid commented on package.json:
```diff
@@ -1,7 +1,9 @@
     "start": "func host start --verbose",
-    "lint": "eslint 'src/**/*.js' 'migrations/**/*.js' '*.js'"
+    "lint": "eslint 'src/**/*.js' 'migrations/**/*.js' '*.js'",
+    "setupDev": "./scripts/setupDev.js",
```
this doesn't work on windows :( 

Should we instead just do `node ./scripts/setupDev.js` since node shebang and windows don't agree ?

craig-cogdill replied:
Thanks for the review and suggestion @AnEmortalKid. I will update this PR shortly

AnEmortalKid reviewed (approved):
Tried the rest of the instructions and they worked, save for a small windows issue.

craig-cogdill commented:
@AnEmortalKid I updated the `package.json` option to use `node` for calling the data fetch script.

@GUI migrated the app to use `nuxt` instead of `eleventy`, so I removed the serve option from my PR. I am not sure the order in which the new `nuxt` calls should happen, so I haven't updated the README with those yet. Haven't had a change to test them in my Linux dev environment yet, so I will update the README after I've had the chance to verify the command order that works.

GUI commented:
@craig-cogdill: Thanks for this! And sorry for throwing a wrench in things by switching around the frontend. This looks super handy, though, so I'll try to take what you have here and update it for the new site tomorrow. Thanks!

rotywonka commented on README.md:
```diff
@@ -13,13 +13,8 @@ This process is clunky right now, so my apologies.
-   - `curl -o site/api/v0/stores/CO/walmart.json https://www.vaccinespotter.org/api/v0/stores/CO/walmart.json`
-4. To run the development server for the website: `yarn run eleventy --serve`. The development site should then be available at http://localhost:8080/.
+3. Fetch data from the [API](https://www.vaccinespotter.org/api/) for the website to run: `yarn run setup`
+4. To run the development server for the website: `yarn run start`. The development site should then be available at http://localhost:8080/.
```
I followed these instructions but got the message that I don't have a dist/ folder. So I had to run `yarn generate` to generate the dist/ folder. Useful to add into the README

craig-cogdill replied:
Thanks @rotywonka, I will update the README with this info

craig-cogdill commented:
@GUI I updated this PR to use the new API routes and static locations specified in your updated README. If you could review my new steps for setup, I'd appreciate it!

patrickdemers6 commented:
Could we add the option to only pull specified states?  In most situations, it's kind of unnecessary to download all states.

craig-cogdill commented:
I will defer that to another PR. I agree it can be unnecessary to download all of the state data, but doing so is easier than specifying individual states, and the total data downloaded is only 16MB. I would be happy to review a PR if you wanted to put one up to my branch though.

patrickdemers6 commented:
Definitely agree it's easiest and that's best. 16MB is very little so nothing to worry about in my opinion!

patrickdemers6 commented on README.md:
```diff
@@ -13,12 +13,10 @@ This process is clunky right now, so my apologies.
-   - `curl -o website/static/api/v0/states/CO.json https://www.vaccinespotter.org/api/v0/states/CO.json`
-   - `curl -o website/static/api/v0/states/CO/postal_codes.json https://www.vaccinespotter.org/api/v0/states/CO/postal_codes.json`
-4. To run the development server for the website: `yarn run dev`. The development site should then be available at http://localhost:3000/.
+3. Fetch data from the [API](https://www.vaccinespotter.org/api/) for the website to run: `yarn setup`
```
Would making the end of this read "for the website by running `yarn setup`" make more sense?

GUI commented:
Sorry this took so long to merge in, but this is great! Thank you!
